src/gov/nist/javax/sip/stack/SIPDialog.java

```diff
@@ -1914,6 +1914,7 @@ public boolean addTransaction(SIPTransaction transaction) {
                 && !firstTransactionId.equals(transaction.getBranchId())
                 && transaction.getMethod().equals(firstTransactionMethod)) {
             setReInviteFlag(true);
+            ackProcessed = false;
```

Comment:
if you dont mind, adding a testCase would be great to ensure no regressions are present in the future..